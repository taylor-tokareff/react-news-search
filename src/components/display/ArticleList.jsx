import React from 'react';
import PropTypes from 'prop-types';

const ArticleList = ({ query }) => (
  <>
    <div>You searched for {query}</div>

  </>
);

ArticleList.propTypes = {
  query: PropTypes.string.isRequired

};

export default ArticleList;
