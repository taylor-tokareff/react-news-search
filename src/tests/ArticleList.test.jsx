/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render } from '@testing-library/react';
import ArticleList from '../components/display/ArticleList';

const TestArticleList = [
  {
    title: 'Test title 1',
    description: 'Test description 1',
    author: 'Test author 1',
    url: 'testurl1.com',
    image: 'testimage1.com'
  },
  {
    title: 'Test title 2',
    description: 'Test description 2',
    author: 'Test author 2',
    url: 'testurl2.com',
    image: 'testimage2.com'
  },
];


describe('ArticleList component', () => {
  it('renders ArticleList', () => {
    const { asFragment } = render(
      <ArticleList articles={TestArticleList}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
