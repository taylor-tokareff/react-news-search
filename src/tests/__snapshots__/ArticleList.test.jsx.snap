// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList component renders ArticleList 1`] = `
<DocumentFragment>
  <div>
    You searched for 
  </div>
  <ul>
    <li>
      <div>
        <p>
          Test title 1
        </p>
        <p>
          Test author 1
        </p>
        <p>
          Test description 1
        </p>
      </div>
    </li>
    <li>
      <div>
        <p>
          Test title 2
        </p>
        <p>
          Test author 2
        </p>
        <p>
          Test description 2
        </p>
      </div>
    </li>
  </ul>
  ;
</DocumentFragment>
`;
